src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+                return false;
+            };
+
+            final RouteFW route = router.resolve(routeId, this.authorization, filter, wrapRoute);
+            if(route != null)
+            {
+                final long newRouteId = route.correlationId();
+                final long newInitialId = supplyInitialId.applyAsLong(newRouteId);
+                final long newReplyId = supplyReplyId.applyAsLong(newInitialId);
+
+                final MessageConsumer newTarget = router.supplyReceiver(newInitialId);
+                final SocksServerStream socksServerStream = new SocksServerStream(this, newTarget,
+                    newRouteId, newInitialId, newReplyId);
+
+                SocksAddressFW socksAddress = socksCommandRequest.address();
+                String address = formatSocksAddress(socksAddress);
+
+                socksServerStream.doApplicationBegin(newTarget,
+                                                     newInitialId,
+                                                     decodeTraceId,
+                                                     address,
+                                                     socksCommandRequest.port());
+
+                correlations.put(newReplyId, socksServerStream);
+            }
+        }
+
+        private String formatSocksAddress(
+            SocksAddressFW socksAddress)
+        {
+            String address = "";
+            switch (socksAddress.kind())
+            {
+                case SocksAddressFW.KIND_DOMAIN_NAME:
+                    address = socksAddress.domainName().asString();
+                    break;
+                case SocksAddressFW.KIND_IPV4_ADDRESS:
+                    OctetsFW ipRO = socksAddress.ipv4Address();
+                    ipRO.buffer().getBytes(ipRO.offset(), writeBuffer, 0, ipRO.sizeof());
+                    address = ((long) writeBuffer.getByte(0) & 0xffL) + "." +
+                              ((long) writeBuffer.getByte(1) & 0xffL) + "." +
+                              ((long) writeBuffer.getByte(2) & 0xffL) + "." +
+                              ((long) writeBuffer.getByte(3) & 0xffL);
+                    break;
+                case SocksAddressFW.KIND_IPV6_ADDRESS:
+                    break;
+                default:
+                    break;
+            }
+            return address;
+        }
+
+
+        private void onNetworkReset(
+            ResetFW reset)
+        {
+            final long traceId = reset.trace();
+            doNetworkReset(traceId);
+        }
+
+        private void onHandshakeRequest(
+            SocksHandshakeRequestFW handshakeRequest)
+        {
+            if (handshakeRequest.version() != 5)
+            {
+                doNetworkEnd(supplyTraceId.getAsLong());
+            }
+            else if (handshakeRequest.methods()
+                                     .buffer()
+                                     .getByte(SocksAuthenticationMethod.NO_AUTHENTICATION_REQUIRED.value()) == -1)
```

Comment:
Please implement `indexOfByte` instead of assuming `NO_AUTHENTICATION_REQUIRED` is the only authentication method offered.